File: Week Assignment 5 using Python/9_Minimum_Bracket_Reversal.py
def min_rev(s):
    n=len(s); 
    if n%2: return -1
    st=[]
    for c in s:
        if c=='{': st.append(c)
        elif st and st[-1]=='{': st.pop()
        else: st.append('}')
    open_=0
    while st and st[-1]=='{': open_+=1; st.pop()
    close=len(st)
    return (open_+1)//2+(close+1)//2
if __name__=='__main__':
    import sys; print(min_rev(sys.stdin.readline().strip()))

File: Week Assignment 5 using C/9_Minimum_Bracket_Reversal.c
#include <stdio.h>
#include <string.h>
int minRev(const char*s){int n=strlen(s);if(n%2)return -1;char st[100000];int top=-1;for(int i=0;i<n;i++){if(s[i]=='{')st[++top]='{';else if(top!=-1&&st[top]=='{')top--;else st[++top]='}';}int open=0;while(top!=-1&&st[top]=='{'){open++;top--;}int close=top+1;return(open+1)/2+(close+1)/2;}
int main(){char s[100000];fgets(s,100000,stdin);s[strcspn(s,"\n")]=0;printf("%d\n",minRev(s));}